corehq/apps/domain/decorators.py

```diff
@@ -352,6 +353,8 @@ def _inner(request, domain, *args, **kwargs):
                 request.user.otp_device = otp_device
                 request.user.is_verified = lambda: True
                 return fn(request, domain, *args, **kwargs)
+            if(api_key):
+                request.user.is_api_request = True
```

Comment:
Why add this new user flag (and a new condition above) rather than set `request.bypass_two_factor` (and use existing condition)?

Reply:
It appeared from other parts of the code that `bypass_two_factor` was an endpoint specific flag that was set with a different semantic intention, I'll review, though, to see if I'm wrong. I also was really unhappy about adding a super specific field, so would be happy to re-use

Reply:
👍 I searched for other usages of `bypass_two_factor` and found two assignments (`request.bypass_two_factor = True`) and no reads other than the one above in this file. After reviewing other usages, it seems reasonable to use it here too.